spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/DatagramPacketMessageMapper.java

```diff
@@ -142,7 +145,23 @@ public DatagramPacket fromMessage(Message<?> message) throws Exception {
 			buffer.put(bytes);
 			bytes = buffer.array();
 		}
-		return new DatagramPacket(bytes, bytes.length);
+		return getDatagramFromHeader(message, bytes, bytes.length);
+	}
+
+	private DatagramPacket getDatagramFromHeader(Message<?> message, byte[] bytes,
+												 int length) throws UnknownHostException {
+		InetAddress inetAddress = null;
+		if (message.getHeaders().get(IpHeaders.IP_ADDRESS, String.class) != null) {
+			inetAddress = InetAddress.getByName(message.getHeaders()
+					.get(IpHeaders.IP_ADDRESS, String.class));
```

Comment:
Would be better to make the call to the `get()` only once. I mean extract variable for the target header value.
Really not sure if compiler makes an optimization here.
From other side we may change in the future the first expression but can forget to change another one.

Although looking at this one more time, I'd say that we should make this `header mapping` as optional.
We recently have faced an issue with Kafka adapters, when the headers from incoming message unconditionally were used to send message: https://github.com/spring-projects/spring-integration-kafka/pull/83

So, how about the `enableHeaderRouting` here as well? 
(Of course, with some other more appropriate name...)

CC. @garyrussell